step-framework-parent/step-framework-timeseries/src/main/java/step/core/timeseries/TimeSeriesBuilder.java

```diff
@@ -0,0 +1,64 @@
+package step.core.timeseries;
+
+import step.core.timeseries.ingestion.TimeSeriesIngestionPipeline;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class TimeSeriesBuilder {
+	
+	private List<TimeSeriesCollection> handledCollections = new ArrayList<>();
+	private Map<Long, TimeSeriesCollection> collectionsByResolution = new TreeMap<>();
+	
+	public TimeSeriesBuilder registerCollection(TimeSeriesCollection collection) {
+		long resolution = collection.getResolution();
+		if (collectionsByResolution.containsKey(resolution)) {
+			throw new IllegalArgumentException("Resolution already present: " + resolution);
+		}
+		handledCollections.add(collection);
+		collectionsByResolution.put(resolution, collection);
+		validateResolutions();
+		return this;
+	}
+	
+	/**
+	 * Each pipeline must have a resolution multiplier of the one before.
+	 */
+	private void validateResolutions() {
+		List<Long> sortedResolutions = new ArrayList<>(collectionsByResolution.keySet());
+		for (int i = 1; i < sortedResolutions.size(); i++) {
+			Long previousResolution = sortedResolutions.get(i - 1);
+			Long currentResolution = sortedResolutions.get(i);
+			if (previousResolution <= currentResolution || currentResolution % previousResolution != 0) {
+				throw new IllegalArgumentException("Invalid resolution: " + currentResolution);
```

Comment:
would be nice to have more details on the errors if it is because of a multiplier issue